### ISSUE TITLE
Should handle fatal errors rather than crashing

### ISSUE THREAD

**Issue by sampart** [whiteoctober/QueueBundle#14]
See http://stackoverflow.com/questions/277224/how-do-i-catch-a-php-fatal-error

It should set the status of the database row to -1 (error), assuming we can get hold of enough information  to do this?

**Comment by richsage** [comment]
what if the fatal error was thrown during doctrine usage.... :-)

**Comment by sampart** [comment]
yeah, I don't actually know how practical this idea is (I've never used `register_shutdown_function` before), but I wanted to record this issue somewhere!
